subsys/bluetooth/mesh/gen_battery_cli.c

```diff
@@ -8,10 +8,16 @@
 #include <bluetooth/mesh/gen_battery_cli.h>
 #include "model_utils.h"
 
-static void decode_status(struct net_buf_simple *buf,
+static int decode_status(struct net_buf_simple *buf,
 			  struct bt_mesh_battery_status *status)
 {
-	status->battery_lvl = net_buf_simple_pull_u8(buf);
+	uint8_t battery_lvl = net_buf_simple_pull_u8(buf);
+
+	if ((battery_lvl >= 0x65) && (battery_lvl <= 0xFE)) {
```

Comment:
This would make the check a little clearer: 
```suggestion
	if ((battery_lvl > 100) && (battery_lvl != BT_MESH_BATTERY_LVL_UNKNOWN)) {
```

Reply:
Yes, thank you! It looks more comprehensible.